### DISCUSSION TITLE
Microsoft Entra not working with Open WebUI

### DISCUSSION THREAD

**Discussion by cwthomas-llu** [open-webui/open-webui#7982] [General]
Hello. I setup SSO auth as per the instructions on https://docs.openwebui.com/features/sso . The env variables MICROSOFT_CLIENT_ID, MICROSOFT_CLIENT_SECRET, and MICROSOFT_CLIENT_TENANT_ID are set. Once I restart the docker container, the "Login via Microsoft" option shows up. When I use it, I can successfully authenticate with Microsoft, but once I am redirected back to open-webui, I get a message that the username or password is incorrect.

```json
{"detail":"The email or password provided is incorrect. Please check for typos and try logging in again."}
```

While checking the logs in the docker container, I see the following error:

```
WARNI [open_webui.utils.oauth] OAuth callback error: invalid_request: AADSTS50146: This application is required to be configured with an application-specific signing key. It is either not configured with one, or the key has expired or is not yet valid. Trace ID: b9b6af1c-402f-4a91-81a5-cf67c256a900 Correlation ID: 70b29c14-73c0-43c3-a545-e6b57acb09fd Timestamp: 2024-12-20 21:23:04Z
```
I have made sure that I have the latest version of the docker image. Has anyone else successfully setup open-webui with Microsoft Entra? What am I doing wrong setting this up? Is there a step missing from the documentation?

**Comment by iunknown80** [comment]
1. Confirm that your newly created key is not expired and is currently within its valid date range
2. Verify if you have added appropriate API permission in Entra.
Entra > App registrations > API permission > + Add a permission > Microsoft Graph > Delegated permission. 

Add email, openid and profile permission. 

See if this solves your issue

**Reply by rhyswynn** [reply]
I have the same issue. I confirmed the delegated permissions for email, openid, and profile are enabled for my user.

**Reply by robbiekouwenberg** [reply]
we also had to set MICROSOFT_OAUTH_SCOPE to "openid email profile"

check your app registration manifest on whether `acceptMappedClaims` is set to `true`.
setting `accessTokenAcceptedVersion` to `2` might also help.

also you are using a single- or multi-tenant app. for multi tenant apps you want app-specific signing keys
see https://learn.microsoft.com/en-us/entra/identity-platform/jwt-claims-customization#security-considerations

**Reply by rhyswynn** [reply]
I added the environment variable you mentioned above to docker compose without any "". I did not have email set on my user, but now I have that set and it's giving me a 403 forbidden in the logs, and the error on the page says 

You do not have permission to access this resource. Please contact your administrator for assistance

accessTokenAcceptedversion isn't in my manifest, and adding it makes the manifest change many other values quite substantially, and doesn't seem to help. I tried updating requestedAccessTokenVersion to 2, but that made no difference either.

**Reply by rhyswynn** [reply]
I also needed to add the ENABLE_OAUTH_SIGNUP=true, now it looks good, thank you so much!

**Comment by zzyco** [comment]
I tried to scale it in k8s. It won't work unless the replica is 1.

I did all the above still having this issue.

**Reply by zzyco** [reply]
nvm. I found out the pods do not have WEBUI_SECRET_KEY set... need to keep them the same value...
